### ISSUE TITLE
Ability to cancel data upload

### ISSUE THREAD

**Issue by vchendrix** [NCEAS/metacatui#564]
In the instances when large data files take a while to upload. Is there a way to allow users to cancel an upload if they just want to submit the metadata instead?

**Comment by laurenwalker** [comment]
Yes, we could add this feature. We can [abort](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/abort) the XMLHttpRequest and remove the data file from the package. The UI can remove the saving styling and allow the user to click the Submit Dataset button. Depending on the state of the XHR, the data file may get saved to Metacat anyway, but if it is not aggregated by the package, it won't show up anywhere in the UI.

**Comment by csjx** [comment]
Yeah, in our original mockups of the menu for each row, we put a cancel button in.  And like Lauren said, if we abort the XHR, we can effectively stop the upload.  On the Metacat side, if the client aborts any time before the last byte is transferred, the mime-multipart parse will get an exception, and so the partial bytes won't be written to `/var/metacat/data`.

**Comment by laurenwalker** [comment]
After thinking about this a bit more...

We may instead want to implement this as an overall Cancel function, where the Cancel button would be placed next to the Submit/Save button and would cancel all current uploads. 

We probably also want to put more upload information in that bottom toolbar (next to the Save button), such as `Uploading 1 of 30 files...` so that the table doesn't have to rely on sorting. I know at some point we want to allow custom sorting of the files in the table, so we will want to think ahead about how to display information about files at the bottom.

**Comment by robyngit** [comment]
The [data package refactor](https://github.com/NCEAS/metacatui/releases/tag/2.39.0) supports cancelling package and eager uploads, but the editor still has no Cancel control. Save is disabled while eager file uploads are running, and the uploading row is read only.
